### ISSUE TITLE
Negative objectness_logits when using RPN in modelzoo

### ISSUE THREAD

**Issue by zherlock030** [facebookresearch/detectron2#1538]
# Instructions To Reproduce the Issue:
I download the model rpn_R_50_FPN_1x in the modelzoo, and use it for inference. 

```
cfg = get_cfg()
cfg.merge_from_file("/xxx/InstanceDetection/detectron2/configs/COCO-Detection/rpn_R_50_FPN_1x.yaml")   # load values from a file
cfg.merge_from_list(["MODEL.WEIGHTS", "/xxx/InstanceDetection/detectron2/pre_train_model/model_final_02ce48.pkl","MODEL.DEVICE","cpu"]) 
predictor = DefaultPredictor(cfg)
im = cv2.imread(fname)
output = predictor(im)
```
then I got the output proposals as:
`{'proposals': Instances(num_instances=1174, image_height=480, image_width=640, fields=[proposal_boxes: Boxes(tensor([[362.3300,  36.7702, 640.0000, 381.5275],
        [172.1298, 102.4427, 603.4216, 480.0000],
        [313.6732,  70.9355, 639.3231, 430.3071],
        ...,
        [  0.0000, 154.4119,  64.0865, 480.0000],
        [111.2227, 308.2463, 354.8970, 480.0000],
        [425.8390,   0.0000, 640.0000,  88.3261]])), objectness_logits: tensor([ 7.9022,  7.7610,  7.5176,  ..., -5.5537, -6.7414, -6.8066])])}`

#my question
I'm not familiar to RPN, my question is why some objectness_logits are negative? I thought the RPN returns proposals that have a positive object probability. 
In the docs it says 
`pred_objectness_logits: predicted objectness scores in [-inf, +inf]; use
        sigmoid(pred_objectness_logits) to estimate P(object).`
I would still think negative object logits will lead to a negative proposal in binary classification, and this should not be returned.
So if it just returns all proposals to the next stage or the negative object logits can still lead to a positive proposal.

**Comment by bluesky314** [comment]
Logits are not probabilities, they are scores which can be negative. Apply softmax to on logits to get probabilities.

**Comment by ppwwyyxx** [comment]
As the doc says: use sigmoid(pred_objectness_logits) to estimate P(object). Probability can never be negative.

**Comment by zherlock030** [comment]
> As the doc says: use sigmoid(pred_objectness_logits) to estimate P(object). Probability can never be negative.

Hey, I know this means all proposals have positive probabilities. According to my knowledge, RPN works as follows:
1. generate many regions
2. filter some negative proposals using softmax/sigmoid
And now my question is, I would assume proposals with negative logits will have a probability below 0.5 using sigmoid, and they should be classified as negative in binary classification. 
And I'm confusing about these details, will all the  proposals be classified using classification network in the next stage in faster rcnn?

**Comment by ppwwyyxx** [comment]
> they should be classified as negative in binary classification.

Not necessarily. There isn't a threshold of "0.5" that decide what is positive or negative, although someone might want to try it this way.

**Comment by zherlock030** [comment]
> > they should be classified as negative in binary classification.
> 
> Not necessarily. There isn't a threshold of "0.5" that decide what is positive or negative, although someone might want to try it this way.

thx for your help, I will check this.
